jetty-core/jetty-session/src/main/java/org/eclipse/jetty/session/AbstractSessionManager.java

```diff
@@ -1350,26 +1350,31 @@ else if (session.getId().equals(getSessionIdManager().getId(id)))
                     }
 
                     throw new BadMessageException(duplicateSession(
-                        requestedSessionId, true, requestedSessionIdFromCookie,
-                        id, true, i < cookieIds));
+                        requestedSessionId, true, requestedSessionIdFrom,
+                        id, true, getRequestedSessionIdFrom(i, cookieIds)));
                 }
                 else if (LOG.isDebugEnabled())
                 {
                     LOG.debug(duplicateSession(
-                        requestedSessionId, true, requestedSessionIdFromCookie,
-                        id, false, i < cookieIds));
+                        requestedSessionId, true, requestedSessionIdFrom,
+                        id, false, getRequestedSessionIdFrom(i, cookieIds)));
                 }
             }
         }
 
-        return new RequestedSession((session != null && session.isValid()) ? session : null, requestedSessionId, requestedSessionIdFromCookie);
+        return new RequestedSession((session != null && session.isValid()) ? session : null, requestedSessionId, requestedSessionIdFrom);
+    }
+
+    private static String getRequestedSessionIdFrom(int index, int cookieIds)
```

Comment:
I think the method name is confusing, it looks like it should be returning a session id, not the the source of the session.  Maybe call it `getRequestedSessionIdSource()`?